libraries/types/transaction/src/transaction.cpp

```diff
@@ -59,29 +59,25 @@ Transaction::Transaction(const dev::RLP &_rlp, bool verify_strict, const h256 &h
 }
 
 trx_hash_t const &Transaction::getHash() const {
-  if (!hash_initialized_) {
-    std::unique_lock l(hash_mu_.val, std::try_to_lock);
-    if (!l.owns_lock()) {
-      l.lock();
-      return hash_;
+  if (!hash_initialized_.load()) {
+    std::unique_lock l(hash_mu_);
+    if (!hash_initialized_.load()) {
```

Comment:
I think inner  `if (!hash_initialized_.load())` could be remove rather than outside one